packages/xrpl/src/models/transactions/MPTokenIssuanceCreate.ts

```diff
@@ -150,12 +155,25 @@ export function validateMPTokenIssuanceCreate(
     }
   }
 
-  const MAX_TRANSFER_FEE = 50000
+  // eslint-disable-next-line @typescript-eslint/consistent-type-assertions -- Not necessary
+  const flags = tx.Flags as number | MPTokenIssuanceCreateFlagsInterface
+  const isTfMPTCanTransfer =
+    typeof flags === 'number'
+      ? isFlagEnabled(flags, MPTokenIssuanceCreateFlags.tfMPTCanTransfer)
+      : flags.tfMPTCanTransfer ?? false
+
   if (typeof tx.TransferFee === 'number') {
```

Comment:
I would put the flags code inside the if block. And change the conditional check to 'tx.TransferFee != null' since the number check is done before it.

Reply:
Ok, it looks confusing because ideally we're checking if the field is not null. Maybe add in != null with the number check?

Reply:
Ok, it looks confusing because ideally we're checking if the field is not null. Maybe add in != null with the number check?

Reply:
adding the check would also give another lint error. The current condition `typeof tx.TransferFee === 'number' ` already implicitly checks that the field is not null